config/config_test.go

```diff
@@ -35,3 +42,56 @@ func (s *configSuite) TestPrefixed(c *C) {
 	c.Assert(s.tlc.policy("quux"), Equals, s.tlc.prefixed(rootPolicy, "quux"))
 	c.Assert(s.tlc.volume("foo", "bar", "quux"), Equals, s.tlc.prefixed(rootVolume, "foo", "bar", "quux"))
 }
+
+func (s *configSuite) TestDumpTarball(c *C) {
+	key := "/volplugin/foo"
+	value := "baz"
+
+	exec.Command("/bin/sh", "-c", "etcdctl set "+key+" "+value).Run()
```

Comment:
It might be a good idea to split the code into two separate functions, one which goes and fetches the data from etcd and one which produces the tarball. This would make it easier to add more tests and it'd also make it better when it comes to adding passing/failing test cases.